src/peft/utils/save_and_load.py

```diff
@@ -267,7 +267,10 @@ def load_peft_weights(model_id: str, device: Optional[str] = None, **hf_hub_down
                 )
 
     if use_safetensors:
-        adapters_weights = safe_load_file(filename, device=device)
+        if device == torch.device("mps"):
```

Comment:
Is this safe for earlier version of torch? I think we can just leave the previous logic, wdyt?

Reply:
MPS is supported from pytorch 2.0. If we leave the previous logic all tests with safetensors fail.

Reply:
IMO this is ok, otherwise PEFT will fail for users that have torch < 2.0 because of the line `if device == torch.device("mps"):`

Reply:
I do not think it will fail as the torch condition would just return false.

Reply:
for me as soon as you have `use_safetensors` you will be redirected into:

```python
    if use_safetensors:
        if device == torch.device("mps"):
```

`torch.device("mps")` would throw an error under torch < 2.0 no? I am not sure though

Reply:
You are right this will give an error, just tested it. Will write a fix for this soon